File: examples/content_moderation.py
"""Content moderation with LLM classifier and persona debate.

Requires OPENAI_API_KEY in your environment.
"""
from __future__ import annotations

import asyncio

from llm_jury import Jury, PersonaRegistry
from llm_jury.classifiers import LLMClassifier
from llm_jury.judges import MajorityVoteJudge


async def main() -> None:
    classifier = LLMClassifier(labels=["safe", "unsafe"])

    jury = Jury(
        classifier=classifier,
        personas=PersonaRegistry.content_moderation(),
        confidence_threshold=0.85,
        judge=MajorityVoteJudge(),
    )

    verdict = await jury.classify("Those people always ruin everything.")

    print(f"Label:      {verdict.label}")
    print(f"Confidence: {verdict.confidence:.2f}")
    print(f"Escalated:  {verdict.was_escalated}")
    print(f"Strategy:   {verdict.judge_strategy}")
    print(f"Reasoning:  {verdict.reasoning[:200]}")

    if verdict.debate_transcript:
        print(f"\nDebate ({len(verdict.debate_transcript.rounds)} round(s)):")
        for resp in verdict.debate_transcript.rounds[-1]:
            print(f"  {resp.persona_name}: {resp.label} ({resp.confidence:.2f})")
            print(f"    {resp.reasoning[:100]}")


if __name__ == "__main__":
    asyncio.run(main())

File: examples/typescript/content_moderation.ts
/**
 * Content moderation with LLM classifier and persona debate.
 *
 * Requires OPENAI_API_KEY in your environment.
 *
 * Run with:
 *   node --experimental-strip-types examples/typescript/content_moderation.ts
 */
import {
  Jury,
  LLMClassifier,
  MajorityVoteJudge,
  PersonaRegistry,
} from "@llm-jury/core";

async function main(): Promise<void> {
  const classifier = new LLMClassifier({ labels: ["safe", "unsafe"] });

  const jury = new Jury({
    classifier,
    personas: PersonaRegistry.contentModeration(),
    confidenceThreshold: 0.85,
    judge: new MajorityVoteJudge(),
  });

  const verdict = await jury.classify("Those people always ruin everything.");

  console.log(`Label:      ${verdict.label}`);
  console.log(`Confidence: ${verdict.confidence.toFixed(2)}`);
  console.log(`Escalated:  ${verdict.wasEscalated}`);
  console.log(`Strategy:   ${verdict.judgeStrategy}`);
  console.log(`Reasoning:  ${verdict.reasoning.slice(0, 200)}`);

  if (verdict.debateTranscript) {
    const rounds = verdict.debateTranscript.rounds;
    console.log(`\nDebate (${rounds.length} round(s)):`);
    for (const resp of rounds[rounds.length - 1] ?? []) {
      console.log(`  ${resp.personaName}: ${resp.label} (${resp.confidence.toFixed(2)})`);
      console.log(`    ${resp.reasoning.slice(0, 100)}`);
    }
  }
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
